src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -496,54 +494,28 @@ private SearchResult CreateSearchResult(SearchOptions searchOptions, IEnumerable
                     .Distinct()
                     .ToList();
 
-                // partition the references to avoid creating an excessively large query
-                foreach (IEnumerable<ResourceTypeAndId> batchOfReferencesToInclude in referencesToInclude.TakeBatch(100))
+                foreach (IEnumerable<ResourceTypeAndId> resourceTypeAndIds in referencesToInclude.TakeBatch(maxIncludeCount))
                 {
-                    // construct the expression typeAndId = <Include1Type, Include1Id> OR  typeAndId = <Include2Type, Include2Id> OR ...
-
-                    SearchParameterExpression expression = Expression.SearchParameter(
-                        _typeIdCompositeSearchParameter,
-                        Expression.Or(batchOfReferencesToInclude.Select(r =>
-                            Expression.And(
-                                Expression.Equals(FieldName.TokenCode, 0, r.ResourceTypeName),
-                                Expression.Equals(FieldName.TokenCode, 1, r.ResourceId))).ToList()));
-
-                    var includeSearchOptions = new SearchOptions
-                    {
-                        Expression = expression,
-                        MaxItemCount = maxIncludeCount,
-                        Sort = Array.Empty<(SearchParameterInfo searchParameterInfo, SortOrder sortOrder)>(),
-                    };
-
-                    QueryDefinition includeQuery = _queryBuilder.BuildSqlQuerySpec(includeSearchOptions);
+                    // issue the requests in parallel
+                    var tasks = resourceTypeAndIds.Select(r => _fhirDataStore.GetAsync(new ResourceKey(r.ResourceTypeName, r.ResourceId), cancellationToken)).ToList();
 
-                    (IReadOnlyList<FhirCosmosResourceWrapper> results, string continuationToken) includeResponse = default;
-                    do
+                    foreach (Task<ResourceWrapper> task in tasks)
```

Comment:
Would responding to a task when it finishes, instead of waiting for the tasks to finish in order, help performance? Something like [this](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/start-multiple-async-tasks-and-process-them-as-they-complete).

Reply:
Good suggestion RB

Reply:
@LTA-Thinking, I think we'll take the original change first. After thinking about it, there may be an implicit order applied (i.e. lastUpdated) which may yield slightly more consistent results